### PR TITLE
Address #364: catch errors per-section

### PR DESCRIPTION
**Pull request by cceckman**
In #364, I accidentally produced a file that (apparently) has empty
section filenames, as understood by the extension. This lead to an
undefined property access (`split`) when trying to work with the file
name; that exception propagated all the way up the rendering stack,
resulting in _no_ coverage being displayed.

With this change, if we encounter an error in `checkSection`, we just return
`false` - if it's an erroneous section, it doesn't apply to the current
file. (This is especially true if there is no `section.file` property!)

This way, we still show coverage wherever we can.

### CODE REVIEW

File: src/coverage-system/sectionfinder.ts [hunk 1]
```diff
@@ -60,15 +60,22 @@ export class SectionFinder {
...
+            // IE: /var/www/ -> /home/me/
+            const sectionFileName = this.resolveFileName(section.file);
+            if (!isPathAbsolute(sectionFileName)) {
+                return this.checkSectionRelative(sectionFileName, editorFileRelative);
+            } else {
+                return this.checkSectionAbsolute(sectionFileName, editorFileRelative, workspaceFolderName);
+            }
+        } catch (error) {
+            const checkSectionError = `[${Date.now()}][renderer]: ignoring section: error from check: ${error}`;
+            this.outputChannel.appendLine(checkSectionError);
```
Comment (ryanluker) [inline src/coverage-system/sectionfinder.ts:77]: nice job finding the logger and logging this!

File: src/coverage-system/sectionfinder.ts [hunk 2]
```diff
@@ -60,15 +60,22 @@ export class SectionFinder {
      * @param section data section to check against filename
      * @param editorFileRelative normalized relative path (against workspace folder) of editor filename, starts with ###
      * @param workspaceFolderName workspace folder name
+     * @returns true iff this section matches (applies to) the provided editorRelativeFile
```
Comment (ryanluker) [inline src/coverage-system/sectionfinder.ts:63]: super minor: `true iff` -> `true if`

File: src/coverage-system/sectionfinder.ts [hunk 3]
```diff
@@ -60,15 +60,22 @@ export class SectionFinder {
...
      */
     private checkSection(section: Section, editorFileRelative: string, workspaceFolderName: string): boolean {
-        // Check if we need to swap any fragments of the file path with a remote fragment
-        // IE: /var/www/ -> /home/me/
-        const sectionFileName = this.resolveFileName(section.file);
-        if (!isPathAbsolute(sectionFileName)) {
-            return this.checkSectionRelative(sectionFileName, editorFileRelative);
-        } else {
-            return this.checkSectionAbsolute(sectionFileName, editorFileRelative, workspaceFolderName);
+        try {
```
Comment (ryanluker) [inline src/coverage-system/sectionfinder.ts:66]: I think this try catch makes sense here, the only other option is down lower in the check functions or higher in the callers of the `checkSection`.

**Review by ryanluker — APPROVED** [review]
One minor thing and then ready to merge!

File: src/coverage-system/sectionfinder.ts (hunk 2 above)
Comment (cceckman) [reply src/coverage-system/sectionfinder.ts]: Done.

**Comment by cceckman** [comment]
Thanks for the comments - glad I wasn't too far off base!

**Review by ryanluker — APPROVED** [review]

**Pull request merged**